sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -388,18 +388,13 @@ case class ArraySort(
     checkArgumentDataTypes() match {
       case TypeCheckResult.TypeCheckSuccess =>
         argument.dataType match {
-          case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
+          case ArrayType(_, _) =>
```

Comment:
I don't know this code well, but is it possible to check it's orderable only if there is no lambda? or check if the lambda's result is orderable if present instead? It does make sense to not fail this case but just wondering if we can preserve a better error message

Reply:
Hi @srowen ,
currently this error message is never shown in the cases where it is relevant.  As the error message on `LessThan` (see above) takes precedence.

I did some experiments but never got it right.  I ran into two problems:
* `checkInputDataTypes` is always called before the `checkInputDataTypes` on the `array_sort` or the lambda function.
* extending `LambdaFunction` is not possible because it is a `case class` and the `higherOrderFunctions` can only work with `LambdaFunction` objects.

If somebody has an idea on how to show the error before `checkInputDataTypes` of `LessThan` is hit I am happy to try it out.

Steven

Reply:
Added extra comment at defaultComparator to explain.